### DISCUSSION TITLE
Accessing Sprite Layer Data

### DISCUSSION THREAD

**Discussion by ghost** [ScratchAddons/ScratchAddons#4705] [Help] [answered]
I've been digging through the addons APIs and Scratch's source code, other addons source code, addons docs, and I just can't find anything.

I'm trying to build an addon the is like the variable manager, but it manages layers instead. I found this
`getLayerOrder () {
        if (this.renderer) {
            return this.renderer.getDrawableOrder(this.drawableID);
        }
        return null;
  }`
at line 836 of https://github.com/LLK/scratch-vm/blob/develop/src/sprites/rendered-target.js, but I don't know how to get the API to import this file vs the virtual machine JS file in the directory below this. If anyone knows a solution to this, help would be greatly appreciated.

**Comment by CST1229** [comment, marked as answer]
I think you can use `target.getLayerOrder()`, with `target` being a sprite or the stage (the items of `vm.runtime.targets`).

**Reply by ghost** [reply]
Thanks, I'll try it out
